src/Modules/Render/ViewScene.cc

```diff
@@ -229,56 +237,30 @@ void ViewScene::syncMeshComponentFlags(const std::string& connectedModuleId, Mod
 
 void ViewScene::execute()
 {
-  /*
-  // hack for headless viewscene. Right now, it hangs/crashes/who knows.
+  fireTransientStateChangeSignalForGeomData();
+  auto state = get_state();
+  auto screenShotMutex = state->getTransientValue(Parameters::VSMutex);
+  auto mutex = transient_value_cast<Mutex*>(screenShotMutex);
+  mutex->lock();
```

Comment:
Better way to lock/unlock a mutex around a scope: the `Guard` class (also in `Mutex.h`) grabs the mutex and locks it, then unlocks in the destructor.

Reply:
Also, with the change to address, this code need not change--it would need to change for unique_ptr though.